### PR TITLE
feat: Hyphenate headings and other large text components

### PR DESCRIPTION
**Pull request by dlnr**
For testing use the word "Carnavalsoptochtvoorbereidingswerkzaamheden" in Heading and Card stories. Make sure the html lang is set to "nl" and it only seems to work if you open the frame of the story.

### CODE REVIEW

File: packages/css/src/common/hyphenation.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+@mixin hyphenation {
+  hyphenate-limit-chars: 12 8 4;
+  hyphens: auto;
+  overflow-wrap: break-word;
+  word-break: break-word;
```
Comment (alimpens) [inline packages/css/src/common/hyphenation.scss:5]: We should use the tokens here

File: proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "amsterdam": {
+    "hyphenation": {
+      "hyphenate-limit-chars": { "value": "12 8 4" },
+      "hyphenate-limit-lines": { "value": "2" },
```
Comment (alimpens) [inline proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json:5]: This is unused

File: packages/css/src/common/hyphenation.scss (hunk 1 above)
Comment (alimpens) [inline packages/css/src/common/hyphenation.scss:5]: `break-word` is deprecated

File: packages/css/src/components/top-task-link/top-task-link.scss [hunk 3]
```diff
@@ -32,6 +34,7 @@
     line-height: var(--amsterdam-top-task-link-label-compact-line-height);
   }
 
+  @include hyphenation;
```
Comment (alimpens) [inline packages/css/src/components/top-task-link/top-task-link.scss:37]: Top task Link has text level 4. Shouldn't we hyphenate all text level 4 elements if Top task needs to be hyphenated?

File: proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "amsterdam": {
+    "hyphenation": {
+      "hyphenate-limit-chars": { "value": "12 8 4" },
+      "hyphens": { "value": "auto" },
+      "overflow-wrap": { "value": "break-word" },
+      "word-break": { "value": "auto-phrase" }
```
Comment (alimpens) [inline proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json:7]: This only works for Japanese? Doesn't seem very necessary imo

Comment (VincentSmedinga) [reply proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json]: No indeed, removed.

File: storybook/storybook-react/config/preview-head.html [hunk 5]
```diff
@@ -1,3 +1,7 @@
 <script>
   window.global = window;
+
+  // Set language to Dutch for Canvas and Stories
+  document.documentElement.setAttribute("lang", "nl");
+  document.querySelectorAll(".docs-story").forEach((e) => e.setAttribute("lang", "nl"));
```
Comment (VincentSmedinga) [inline storybook/storybook-react/config/preview-head.html:6]: This should work, right? Any idea why it doesn’t?

File: proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "amsterdam": {
+    "hyphenation": {
+      "hyphenate-limit-chars": { "value": "12 8 4" },
+      "hyphens": { "value": "auto" },
+      "overflow-wrap": { "value": "break-word" },
```
Comment (VincentSmedinga) [inline proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json]: Should we use `overflow-wrap` or `word-break: break-word`? I think I’d prefer the latter as it fills up the first line as much as possible. See the second example [here on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_text/Wrapping_breaking_text#breaking_long_words). @dlnr Did you investigate this?

File: proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json (hunk 4 above)
Comment (VincentSmedinga) [reply proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json]: I’ve removed it.

File: storybook/storybook-react/config/preview-head.html (hunk 5 above)
Comment (alimpens) [reply storybook/storybook-react/config/preview-head.html]: I think stories are loaded after the document is loaded, so elements with the `docs-story` class don't exist yet on document load. I've added this script to the story `decorators`, that seems to work

File: proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json (hunk 6 above)
Comment (alimpens) [reply proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json]: `word-break: break-word` is deprecated, we shouldn't use that. The example uses `word-break: break-all`. `word-break: break-all` beats `hyphens: auto` though, so we can't really use that either 😅

Comment (dlnr) [reply proprietary/tokens/src/common/amsterdam/hyphenation.tokens.json]: I've tried "auto-phrase" but it did not seem to do anything.
> Has the same effect as word-break: normal except that language-specific analysis is performed to improve word breaks by not placing them in the middle of natural phrases.

File: storybook/storybook-react/config/preview-head.html (hunk 5 above)
Comment (dlnr) [reply storybook/storybook-react/config/preview-head.html]: Does this also work when you open the frame directly?

Comment (alimpens) [reply storybook/storybook-react/config/preview-head.html]: > although it fires too often – it queries the entire document for each Story on it

Derp, missed that. I've gone with another approach, simply wrapping the story in a div with the correct language. Should be more robust.

> Does this also work when you open the frame directly?

Yeah, the new approach does. The decorators also cascade properly, so the stories are wrapped in `<div lang="nl">` first, and any other decorators defined on the story level after.

**Review by VincentSmedinga — APPROVED** [review]

**Pull request merged**